code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -30,6 +30,10 @@
 	var/target_temperature = T20C
 	var/heat_capacity = 0
 	var/interactive = TRUE // So mapmakers can disable interaction.
+	/// The pyroclastic anomaly core inside a Thermomachine
+	var/obj/item/assembly/signaler/anomaly/pyro/pyro_anomaly_core
```

Comment:
You don't clean this reference up. You should make it a weakref.

Reply:
I have a better suggestion. you can just use a boolean var like `is_pyro_core_installed`, to tell if the core is installed or not, when installing the core just move that core to nullspace, delete it, & set this var to true, & during deconstruction check for this var and spawn a core if it's true(indicating it was installed), This was you don't have to hold to refs

Reply:
I do not know what a weakref is. If you tell me how to make the change to it I will do so.